web3swift/src/ERC20/ERC20.swift

```diff
@@ -37,6 +37,13 @@ public class ERC20 {
         }
     }
     
+    public func totalSupply(tokenContract: EthereumAddress, completion: @escaping((Error?, BigUInt?) -> Void)) {
+        let function = ERC20Functions.totalSupply(contract: tokenContract)
+        function.call(withClient: self.client, responseType: ERC20Responses.balanceResponse.self) { (error, balanceResponse) in
```

Comment:
`balanceResponse` here is re-using an existing response type for a different purpose. Maybe we should rename the existing one to be more generic for both use cases (`valueResponse` instead?)

Reply:
Hi Matt, Yes I agree with you, balanceResponse was already used for allowance and balanceOf that's why I also used it for totalSupply but I guess a more generic response could be better. We could also remove decimalsResponse that way as it seems to be a copy of balanceResponse. Also I am not sure about "valueResponse" as it seems maybe too generic. For example nameResponse could be considered as a valueResponse but it returns a string and not a BigUInt. What about "bigUIntResponse" ? At the end, if you prefer that we use valueResponse no problem, was just brainstorming a little bit :)